fv3gfs/util/halo_data_transformer.py

```diff
@@ -0,0 +1,914 @@
+from dataclasses import dataclass
+from .buffer import Buffer
+from .quantity import Quantity
+from .types import NumpyModule
+from typing import List, Tuple, Optional, Dict, Any, Sequence
+from enum import Enum
+from .rotate import rotate_scalar_data, rotate_vector_data
+from .utils import device_synchronize
+import numpy as np
+from uuid import UUID, uuid1
+from .cuda_kernels import (
+    pack_scalar_f64_kernel,
+    pack_vector_f64_kernel,
+    unpack_scalar_f64_kernel,
+    unpack_vector_f64_kernel,
+)
+import abc
+
+try:
+    import cupy as cp
+except ImportError:
+    cp = None
+
+
+@dataclass
+class HaloUpdateSpec:
```

Comment:
Suggest renaming this to `QuantityHaloSpec`, since it does not have the full information needed to perform a halo update.